### ISSUE TITLE
braille characters (used by loader animations) don't show correctly (only macOS)

### ISSUE THREAD

**Issue by voltuer** [microsoft/vscode#174521]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.75.1
- OS Version: macOS 12.6

Steps to Reproduce:

1. Use any CLI utility that displays a loader animation that uses braille, such as `npm install`
2. For every step of the animation, you should see an array of dots that don't use up a full square space, but leave some blank space and look like a "snake" (**this works fine in Windows**)

This is how it looks in other terminals such as iTerm2 or Terminal.app:
<img width="33" alt="image" src="https://user-images.githubusercontent.com/197329/219254964-83b84715-e128-46c3-bdeb-4a35d6c8a92b.png">

This is how it looks in VS Code (i've tried changing fonts, it doesn't help)
<img width="31" alt="image" src="https://user-images.githubusercontent.com/197329/219254781-a3dcc6b9-a2b3-4fdd-800b-eb8c88ce4ef5.png">

This is the character:
```
⠏
```

On Windows version of VS Code, this behaves correctly

on macOS, instead of white space it shows circles with transparent filling

**Comment by sebolio** [comment]
any news?

**Comment by sebolio** [comment]
Hello, it's been half a year and it's been tagged many things, I'd love to know if this is being taked into consideration since it's very bad UX

**Comment by meganrogge** [comment]
Interesting that this works on windows but not mac.... 

This issue reproduces in the monaco editor

**Comment by meganrogge** [comment]
it looks like for xterm.js we are using the python3.9 spinners https://insiders.vscode.dev/github/meganrogge/xterm.js/blob/merogge/acc-tree/.venv/lib/python3.9/site-packages/pip/_vendor/rich/_spinners.py#L21-L37

**Comment by meganrogge** [comment]
I found this - have you tried other fonts?

https://github.com/gizak/termui/issues/277

**Comment by sebolio** [comment]
i found out that it's Chromium's fault, the issue happens in Chrome for macOS and every other child browser, but doesn't happen in Firefox nor Safari, so I opened a bug report there

**Comment by pradyunsg** [comment]
![Screen Shot 2024-08-01 at 12 54 20](https://github.com/user-attachments/assets/06916ea9-b76a-47aa-8f23-d96161f602dc)

I just double checked today and it looks like Chrome 127.0.6533.89 does not have the braille characters getting rendered incorrectly anymore.

**Comment by pradyunsg** [comment]
I double checked today and it looks like Chrome Version 127.0.6533.89 renders Braille characters without hollow circles on MacOS.

<img width="952" alt="Screenshot 2024-08-02 at 10 50 29" src="https://github.com/user-attachments/assets/53d458bd-2d8e-481c-94f9-046dd3ac4dfd">

(I have a macOS app that's rendering the top bar differently)

**Comment by pradyunsg** [comment]
So... I spent a bit too long today investigating what was happening here and here's how I've resolved this issue for myself:

```diff
-    "editor.fontFamily": "VictorMono Nerd Font",
+    "editor.fontFamily": "VictorMono Nerd Font, AppleBraille",
```

Before / After
<img width="163" alt="Screenshot 2024-08-07 at 22 22 32" src="https://github.com/user-attachments/assets/03880d4c-5db8-4e68-9377-72201c67e7ef"> <img width="163" alt="Screenshot 2024-08-07 at 22 22 42" src="https://github.com/user-attachments/assets/5b580c85-4d0b-47d9-ad32-a7bb2e95b036">

In case the diagnostic steps are useful to someone, here's the portion of my search tree that resulted in this solution / bodge....

I looked through my macOS font book for which fonts were providing symbols for ⡹ (U+2879) and ended up finding "Apple Braille" and "Apple Symbols". "Apple Symbols" had the wrong amount of whitespace when I sent that as the fallback font and the "Apple Braille" looked like a drop-in replacement.

Turns out, Apple Braille looked different in the OS's Font Book vs in VS Code. The reason for this ended up being that Apple Braille has multiple variants and... for some reason... the version of Chromium being used in VS Code uses either Outline6Dot or Outline8Dot variants by default (these have the hollow/outline circles) instead of the Regular variant which does not have hollow circles. I assume this has to do with the font-renderer-level behaviour.

None the less, placing the `postScriptName` of the specific font variant within the "Apple Braille" family that I want to use was "AppleBraille" and adding that to the stack makes it use the correct-for-me variant of the OS-provided Braille symbols!

**Comment by pradyunsg** [comment]
Alternatively, you can also do what I did recently and change to a font that includes braille characters (like https://typeof.net/Iosevka/). 😅

**Comment by Tyriar** [comment]
I considered creating an issue for this for the editor as well, but the visualize progress isn't as important there and it's more possible people may want the filled/empty circle rendering in the editor to more easily identify which circles are filled.

**Comment by Tyriar** [comment]
This failed the macOS build when merged:

https://dev.azure.com/monacotools/Monaco/_build/results?buildId=290850&view=logs&j=55ac390b-ffb1-50ea-3650-525dd9a3fd80&t=c01b9029-2b08-503e-eba8-b45cf5956c33

**Comment by hazycora** [comment]
This occurs in the editor too, by the way. even still. 

<img width="288" alt="Image" src="https://github.com/user-attachments/assets/518965ce-5c8f-443d-b2a5-162097759b16" />

it's fixed by adding AppleBraille to the font list, same as the fix for this did for the terminal

<img width="290" alt="Image" src="https://github.com/user-attachments/assets/0c10c215-3eae-4660-8785-594e675c8443" />
